extensions/eclipse-che-theia-plugin-ext/src/node/che-client-ip-service.ts

```diff
@@ -0,0 +1,27 @@
+/*********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import * as express from 'express';
+import { BackendApplicationContribution } from '@theia/core/lib/node/backend-application';
+import { injectable } from 'inversify';
+
+@injectable()
+export class CheClientIpServiceContribution implements BackendApplicationContribution {
+
+    configure(app: express.Application): void {
+        app.get('/che/client-ip', (req, res) => {
+            let clientIp = req.connection.remoteAddress;
```

Comment:
wondering will this work for both IPv4 & IPv6?

Reply:
Yes, it will work correct for both https://nodejs.org/api/net.html#net_socket_remoteaddress